[Bug]: e2e tests failf opentofu >= 1.11 with "Error: Unable to Move Resource State"

### What happened?

This seem to be scoped to the rdns plugin framework upgrade tests.

```
=== NAME  TestAccRDNSResource_PrimaryIP_UpgradePluginFramework
    resource_test.go:308: Step 2/2 error: Error running non-refresh plan: exit status 1
        
        Error: Unable to Move Resource State
        
          with hcloud_primary_ip.ipv6,
          on terraform_plugin_test.tf line 11, in resource "hcloud_primary_ip" "ipv6":
          11: resource "hcloud_primary_ip" "ipv6" {
        
        The target resource implementation does not include move resource state
        support. The resource implementation can be updated by the provider
        developers to include this support with the ResourceWithMoveState interface.
        
        Source Provider Address: registry.opentofu.org/hetznercloud/hcloud
        Source Resource Type: hcloud_primary_ip
        Source Resource Schema Version: 0
        Target Resource Type: hcloud_primary_ip
    panic.go:694: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: Unable to Move Resource State
        
          with hcloud_rdns.ipv6,
          on terraform_plugin_test.tf line 17, in resource "hcloud_rdns" "ipv6":
          17: resource "hcloud_rdns" "ipv6" {
        
        The target resource implementation does not include move resource state
        support. The resource implementation can be updated by the provider
        developers to include this support with the ResourceWithMoveState interface.
        
        Source Provider Address: registry.opentofu.org/hetznercloud/hcloud
        Source Resource Type: hcloud_rdns
        Source Resource Schema Version: 0
        Target Resource Type: hcloud_rdns
--- FAIL: TestAccRDNSResource_PrimaryIP_UpgradePluginFramework (12.37s)
```

Example pipeline: https://github.com/hetznercloud/terraform-provider-hcloud/actions/runs/29726441752/job/88300734275?pr=1493

### What did you expect to happen?

The e2e tests not to fail. 

### Please provide a minimal working example

https://github.com/hetznercloud/terraform-provider-hcloud/actions/runs/29726441752/job/88300734275?pr=1493

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -82,7 +82,7 @@ jobs:
       - if: matrix.tool == 'opentofu'
         run: |
           echo TF_ACC_TERRAFORM_PATH="$(which tofu)" >> $GITHUB_ENV
-          echo TF_ACC_PROVIDER_NAMESPACE="hashicorp" >> $GITHUB_ENV
+          echo TF_ACC_PROVIDER_NAMESPACE="hetznercloud" >> $GITHUB_ENV
           echo TF_ACC_PROVIDER_HOST="registry.opentofu.org" >> $GITHUB_ENV
 
       - run: go test -v -timeout=30m -parallel=8 -coverprofile=coverage.txt -coverpkg=./... ./...

